landoapi/projects.py

```diff
@@ -14,6 +14,13 @@
 SEC_PROJ_CACHE_KEY = "secure-project-phid"
 CHECKIN_PROJ_SLUG = "check-in_needed"
 CHECKIN_PROJ_CACHE_KEY = "checkin-project-phid"
+
+# needs-testing-tag is used to mark those revisions that are approved but are
+# missing a testing tag. This is used to trigger a warning when attempting to
+# land something.
+NEEDSTESTINGTAG_PROJ_SLUG = "needs-testing-tag"
```

Comment:
If we check on needs-testing-tag this will also rely on the Herald rules being set up. I'm not necessarily opposed to that, but something that might be more straightforward is checking for the _lack of_ one of the testing tags (`testing-approved`, `testing-exception-elsewhere`, `testing-exception-other`, `testing-exception-ui`, `testing-exception-unchanged`).

Reply:
Was about to reach out RE: preference. Either works, and I do prefer the approach of checking for the lack of any of the tags (i.e. what you suggested), this was easier to test while I was coding things. Is the list of testing tags finalized? (note that changing any of the tags means we will need up update the hardcoded values).

Reply:
The list of testing tags is stable. I don't think "finalized" is the right word, since I could definitely see adding/removing/renaming tags as needs evolve. But the bar to change is going to be fairly high and if it does, we'll have other tooling to update as well so I'm fine with documenting this and landing with the hardcoded list.